### PR TITLE
Error if the 'sdkconfig.defaults' can't be found.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: tools/main.toit [hunk 1]
```diff
@@ -198,17 +198,17 @@ ensure_partitions_ dir/string --toit_root/string --chip/string:
 ensure_sdkconfig_ dir/string --toit_root/string --chip/string:
   if file.is_file "$dir/sdkconfig" or file.is_file "$dir/sdkconfig.defaults": return
 
-  sdk_config_path := toit_sdk_config_path_for_ --chip=chip
+  if file.is_file "$dir/sdkconfig.patch":
+    throw "Variants should only patch sdkconfig.defaults, not sdkconfig."
```
Comment (kasperl) [inline tools/main.toit:202]: Maybe pass the UI in and produce a nicer error?

File: tools/main.toit [hunk 2]
```diff
@@ -198,17 +198,17 @@ ensure_partitions_ dir/string --toit_root/string --chip/string:
...
-  if file.is_file "$toit_root/$sdk_config_path":
-    copy_file --from="$toit_root/$sdk_config_path" --to="$dir/sdkconfig"
-    if file.is_file "$dir/sdkconfig.patch":
-      apply_file_patch_ --patch_path="$dir/sdkconfig.patch" --file_path="$dir/sdkconfig"
   if file.is_file "$toit_root/$sdk_config_defaults_path":
     copy_file --from="$toit_root/$sdk_config_defaults_path" --to="$dir/sdkconfig.defaults"
     if file.is_file "$dir/sdkconfig.defaults.patch":
       apply_file_patch_ --patch_path="$dir/sdkconfig.defaults.patch" --file_path="$dir/sdkconfig.defaults"
+  else:
+    throw "No sdkconfig.defaults found for chip '$chip'."
```
Comment (kasperl) [inline tools/main.toit:211]: Maybe pass the UI in and produce a nicer error?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: tools/main.toit (hunk 2 above)
Comment (floitsch) [reply tools/main.toit]: done.

**Pull request merged**